### DISCUSSION TITLE
How to get list of sub domains of the domain

### DISCUSSION THREAD

**Discussion by sukesh-ak** [MichaCo/DnsClient.NET#135] [Q&A] [answered]
How can we get a list of sub-domains for the main domain being queried?
Main domain - domain.com [works]

sub-domain - abc.domain.com [A record for this not listed in the main domain query]

**Comment by MichaConrad** [comment, marked as answer]
Hi @sukesh-ak 
that's not how DNS works, if that were possible, hackers would have it even easier.

The only query which can give you all the zone information is a zone transfer. You can do that with DnsClient (or any other DNS tool) but you'll need access (permissions) for that query on your DNS server.

**Comment by sukesh-ak** [comment]
> Hi @sukesh-ak
> that's not how DNS works, if that were possible, hackers would have it even easier.
> 
> The only query which can give you all the zone information is a zone transfer. You can do that with DnsClient (or any other DNS tool) but you'll need access (permissions) for that query on your DNS server.

Thanks for the response.
